### PR TITLE
Introduce loader.EventLog helper class

### PR DESCRIPTION
**Pull request by jart**
This class delegates to RecordReader and decodes the individual protos. It
supports marking which is going to be needed when transactions have transient
failures. Instances are sortable, which assists with reading them in order.

### CODE REVIEW

File: tensorboard/loader.py [hunk 1]
```diff
@@ -672,6 +674,169 @@ def _make_red(self, text):
             util.Ansi.RESET)
 
 
+@util.closeable
+@functools.total_ordering
+@six.python_2_unicode_compatible
+class EventLog(object):
```
Comment (wchargin) [inline tensorboard/loader.py]: What do you think about `EventLog` vs. `EventLogReader`?

File: tensorboard/loader.py [hunk 2]
```diff
@@ -672,6 +674,169 @@ def _make_red(self, text):
...
+  """Helper class for reading from event log files.
+
+  This class is wrapper around BufferedRecordReader that operates on
```
Comment (wchargin) [inline tensorboard/loader.py]: nit: is **a** wrapper

File: tensorboard/loader_test.py [hunk 3]
```diff
@@ -246,6 +246,86 @@ def _on_bar(self, format_, *args):
...
+    log.close()
+
+  def testEmptyFile_returnsNoneRecords(self):
+    path = self._save_records('events.out.tfevents.0.localhost', [])
+    with self.EventLog(path) as log:
+      self.assertIsNone(log.get_next_event())
+      self.assertIsNone(log.get_next_event())
+
+  def testReadOneEvent(self):
+    event = tf.Event()
```
Comment (wchargin) [inline tensorboard/loader_test.py]: You can save a few statements everywhere by using `tf.Event(step=123)`.

File: tensorboard/loader_test.py [hunk 4]
```diff
@@ -246,6 +246,86 @@ def _on_bar(self, format_, *args):
...
+    path = self._save_records('events.out.tfevents.0.localhost',
+                              [event1.SerializeToString(),
+                               event2.SerializeToString()])
+    with self.EventLog(path) as log:
+      log.mark()
+      self.assertEqual(event1, log.get_next_event())
+      self.assertEqual(event2, log.get_next_event())
+      log.reset()
+      self.assertEqual(event1, log.get_next_event())
+      with self.assertRaisesRegexp(ValueError, 'monotonic'):
```
Comment (wchargin) [inline tensorboard/loader_test.py]:
```py
with six.assertRaisesRegex(self, ValueError, r'monotonic'):
```

File: tensorboard/loader.py (hunk 1 above)
Comment (jart) [inline tensorboard/loader.py]: Done.

File: tensorboard/loader.py (hunk 2 above)
Comment (jart) [inline tensorboard/loader.py]: Done. Also I don't consider that a nit.

File: tensorboard/loader_test.py (hunk 3 above)
Comment (jart) [inline tensorboard/loader_test.py]: Good idea.

File: tensorboard/loader_test.py (hunk 4 above)
Comment (jart) [inline tensorboard/loader_test.py]: Done.

**Comment by wchargin** [comment]
Still approved.

**Pull request merged**